packages/datadog-plugin-pg/src/index.js

```diff
@@ -7,12 +7,17 @@ class PGPlugin extends DatabasePlugin {
   static get operation () { return 'query' }
   static get system () { return 'postgres' }
 
-  start ({ params = {}, statement, processId }) {
+  start ({ params = {}, pgQuery, processId }) {
     const service = getServiceName(this.config, params)
+    const originalStatement = pgQuery.text
+
+    if (this.config.dbmPropagationMode === 'service') {
```

Comment:
@bengl do we have a one-time-warning helper or anything?

A future value that customers will be able to set is called `all`, and I think if they provide that value we might want to print a warning then fallback to `service`.

Reply:
Though, on second thought, such a call should really happen here:
https://github.com/DataDog/dd-trace-js/pull/2480/files#diff-35b625667f1fbe3cfe5d2d1290484d9643c044d90ece6dea383d9e26a1633755R82

Reply:
Should this condition be in every database plugin? Maybe there could be an additional helper in the base class, something like `pgQuery.text = this.injectDBM(pgQuery.text)` so that when it's disabled it simply returns the string as-is, and then the condition and where to add the comment can be centralized.

Reply:
this has been updated to push the logic up to the db class